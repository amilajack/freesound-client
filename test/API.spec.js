import FreeSound from "../index.ts";

require("dotenv").config();

function removeVariableProperties(obj) {
  /* eslint-disable f */
  delete obj.num_downloads;
  delete obj.count;
  delete obj.results;
  delete obj.bitrate;
  delete obj.id;
  delete obj.average_rating;
  delete obj.duration;
  delete obj.ac_analysis;
  /* eslint-enable */
  return obj;
}

describe("API", function testApi() {
  beforeAll(async () => {
    this.freeSound = new FreeSound();
    this.freeSound.setToken(process.env.CLIENT_SECRET);
    this.freeSound.setClientSecrets(
      process.env.CLIENT_ID,
      process.env.CLIENT_SECRET
    );
  });

  describe("User", () => {
    it("should get user", async () => {
      expect(
        removeVariableProperties(await this.freeSound.getUser("Jovica"))
      ).toMatchSnapshot();
    });

    it("should get a users data", async () => {
      const user = await this.freeSound.getUser("Jovica");
      const [sounds, packs, bookCat, bookCatSounds] = await Promise.all([
        user.sounds(),
        user.packs(),
        user.bookmarkCategories(),
        user.bookmarkCategorySounds(),
      ]);
      expect(sounds).toBeTruthy();
      expect(packs).toBeTruthy();
      expect(bookCat).toBeTruthy();
      expect(bookCatSounds).toBeTruthy();
      expect(removeVariableProperties(sounds)).toMatchSnapshot();
      expect(removeVariableProperties(packs)).toMatchSnapshot();
      expect(removeVariableProperties(bookCat)).toMatchSnapshot();
      expect(removeVariableProperties(bookCatSounds)).toMatchSnapshot();
    });
  });

  describe("Pack", () => {
    it("should get pack", async () => {
      const pack = await this.freeSound.getPack(9678);
      expect(removeVariableProperties(pack)).toMatchSnapshot();
    });

    it("should get pack data", async () => {
      const pack = await this.freeSound.getPack(9678);
      const sounds = await pack.sounds();
      expect(removeVariableProperties(sounds)).toMatchSnapshot();
    });
  });

  describe("Sound", () => {
    it("should get sound", async () => {
      const sound = await this.freeSound.getSound(96541);
      expect(removeVariableProperties(sound)).toBeTruthy();
    });

    it("should get sound data", async () => {
      const sound = await this.freeSound.getSound(96541);
      const [analysis, similar, comments] = await Promise.all([
        sound.getAnalysis(),
        sound.getSimilar(),
        sound.getComments(),
      ]);
      expect(analysis).toBeTruthy();
      expect(similar).toBeTruthy();
      expect(comments).toBeTruthy();
      expect(removeVariableProperties(analysis)).toMatchSnapshot();
    });
  });

  describe("Search", () => {
    it("should text search", async () => {
      const query = "violoncello";
      const page = 1;
      const filter = "tag:tenuto duration:[1.0 TO 15.0]";
      const sort = "rating_desc";
      const fields = "id,name,url";
      const search = await this.freeSound.textSearch(query, {
        page,
        filter,
        sort,
        fields,
      });
      expect(removeVariableProperties(search)).toMatchSnapshot();
    });

    it("should perform combined search", async () => {
      const result = await this.freeSound.combinedSearch({
        target: "rhythm.bpm:120",
        filter: "tag:loop",
      });
      expect(result).toBeTruthy();
      expect(removeVariableProperties(result)).toMatchSnapshot();
    });

    it("should perform content search", async () => {
      const result = await this.freeSound.contentSearch({
        target: "lowlevel.pitch.mean:220",
      });
      expect(result).toBeTruthy();
      expect(removeVariableProperties(result)).toMatchSnapshot();
    });
  });

  describe.skip("Auth", () => {
    it("should go through oauth process", async () => {
      // OAuth login
      await this.freeSound.setToken("your-api-key", "oauth");
      // Set your application's client_id and client_secret
      await this.freeSound.setClientSecrets(
        "your-client-id",
        "your-secret-key"
      );
      // Make the user navigate here
      await this.freeSound.getLoginURL();
      // Use the authorization code from the login
      await this.freeSound.postAccessCode("your-temporary-code-from-login");
    });

    it("should get me", async () => {
      expect(await this.freeSound.me()).toBeTruthy();
    });

    it("should get pending sounds", async () => {
      expect(await this.freeSound.getPendingSounds()).toBeTruthy();
    });
  });
});
