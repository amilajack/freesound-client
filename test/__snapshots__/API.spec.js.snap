// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Pack should get pack 1`] = `
Object {
  "created": "2012-05-23T22:25:41",
  "description": "",
  "download": [Function],
  "id": 9678,
  "name": "Zoo",
  "num_downloads": 32,
  "num_sounds": 2,
  "sounds": [Function],
  "url": "https://freesound.org/people/reinsamba/packs/9678/",
  "username": "reinsamba",
}
`;

exports[`API Pack should get pack data 1`] = `Promise {}`;

exports[`API Sound should get sound 1`] = `
Object {
  "analysis": "No descriptors specified. You should indicate which descriptors you want with the 'descriptors' request parameter.",
  "analysis_frames": "https://freesound.org/data/analysis/96/96541_7037_frames.json",
  "analysis_stats": "https://freesound.org/apiv2/sounds/96541/analysis/",
  "avg_rating": 4.666666666666665,
  "bitdepth": 24,
  "bitrate": 2117,
  "bookmark": "https://freesound.org/apiv2/sounds/96541/bookmark/",
  "channels": 2,
  "comment": "https://freesound.org/apiv2/sounds/96541/comment/",
  "comments": "https://freesound.org/apiv2/sounds/96541/comments/",
  "created": "2010-05-12T20:29:49",
  "description": "For the user \\"Stealth Inc.\\", my girlfriend saying \\"Coming soon\\". Recorded with an AT2020 microphone into an Apogee Duet and edited with Record.",
  "download": "https://freesound.org/apiv2/sounds/96541/download/",
  "duration": 1.8774829932,
  "filesize": 496906,
  "geotag": null,
  "id": 96541,
  "images": Object {
    "spectral_l": "https://freesound.org/data/displays/96/96541_7037_spec_L.jpg",
    "spectral_m": "https://freesound.org/data/displays/96/96541_7037_spec_M.jpg",
    "waveform_l": "https://freesound.org/data/displays/96/96541_7037_wave_L.png",
    "waveform_m": "https://freesound.org/data/displays/96/96541_7037_wave_M.png",
  },
  "license": "http://creativecommons.org/licenses/by/3.0/",
  "name": "coming soon.wav",
  "num_comments": 1,
  "num_ratings": 3,
  "pack": "https://freesound.org/apiv2/packs/6201/",
  "pack_name": "Stealth Inc.",
  "previews": Object {
    "preview-hq-mp3": "https://freesound.org/data/previews/96/96541_7037-hq.mp3",
    "preview-hq-ogg": "https://freesound.org/data/previews/96/96541_7037-hq.ogg",
    "preview-lq-mp3": "https://freesound.org/data/previews/96/96541_7037-lq.mp3",
    "preview-lq-ogg": "https://freesound.org/data/previews/96/96541_7037-lq.ogg",
  },
  "rate": "https://freesound.org/apiv2/sounds/96541/rate/",
  "samplerate": 44100,
  "similar_sounds": "https://freesound.org/apiv2/sounds/96541/similar/",
  "tags": Array [
    "request",
    "female",
    "girl",
    "talk",
    "vocal",
    "voice",
    "american",
    "woman",
    "english",
    "speak",
  ],
  "type": "wav",
  "url": "https://freesound.org/people/tim.kahn/sounds/96541/",
  "username": "tim.kahn",
}
`;

exports[`API User should get a users data 1`] = `
Object {
  "count": 7813,
  "getItem": [Function],
  "getSound": [Function],
  "next": "https://freesound.org/apiv2/users/Jovica/sounds/?page=2",
  "nextPage": [Function],
  "previous": null,
  "previousPage": [Function],
  "results": Array [
    Object {
      "id": 384811,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Trinity Church.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384810,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Warm Bath MW.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384809,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Drama Rama.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384808,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Haunted Hill.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384807,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Classic Strings.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384806,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  ColdPhaze.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384805,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Contact.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384804,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Dawn Of The Dead.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384803,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Additivism.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384802,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Basic Element.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384801,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Blair Witch Project.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384800,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Breathy Choir.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384789,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Phasolina.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384788,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  PercBed.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
    Object {
      "id": 384787,
      "license": "http://creativecommons.org/licenses/by-nc/3.0/",
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer  -  Smallville.wav",
      "tags": Array [
        "chords",
        "chord-sequence",
        "60bpm",
        "sequence",
        "pad",
      ],
      "username": "Jovica",
    },
  ],
}
`;

exports[`API User should get a users data 2`] = `
Object {
  "count": 239,
  "getItem": [Function],
  "getPack": [Function],
  "next": "https://freesound.org/apiv2/users/Jovica/packs/?page=2",
  "nextPage": [Function],
  "previous": null,
  "previousPage": [Function],
  "results": Array [
    Object {
      "created": "2017-03-19T16:21:34",
      "description": "",
      "id": 21603,
      "name": "20170319 -  4 chords in a sequence at 60 BPM  -  Embracer",
      "num_downloads": 16,
      "num_sounds": 32,
      "sounds": "https://freesound.org/apiv2/packs/21603/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21603/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-03-19T16:01:40",
      "description": "",
      "id": 21601,
      "name": "20170319 -  3 melancholic chord sequence at 60 BPM  -  Embracer",
      "num_downloads": 13,
      "num_sounds": 32,
      "sounds": "https://freesound.org/apiv2/packs/21601/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21601/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-03-05T14:07:15",
      "description": "",
      "id": 21528,
      "name": "20170226 - 4 melancholic descending chords at 60 BPM - Embracer",
      "num_downloads": 28,
      "num_sounds": 32,
      "sounds": "https://freesound.org/apiv2/packs/21528/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21528/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-02-05T11:19:06",
      "description": "",
      "id": 21349,
      "name": "20170205 - Embracer - C chord sequence @ 80 bpm - NI Guitar Rig 5",
      "num_downloads": 12,
      "num_sounds": 41,
      "sounds": "https://freesound.org/apiv2/packs/21349/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21349/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-01-20T16:50:57",
      "description": "",
      "id": 21189,
      "name": "20170120 - Embracer - white key slow arp up over 2 octaves @ 140 bpm - NI Guitar Rig 5",
      "num_downloads": 10,
      "num_sounds": 40,
      "sounds": "https://freesound.org/apiv2/packs/21189/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21189/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-01-14T11:44:11",
      "description": "",
      "id": 21147,
      "name": "20170114 - Embracer - white key chord sequence @ 140 bpm - NI Guitar Rig 5",
      "num_downloads": 13,
      "num_sounds": 40,
      "sounds": "https://freesound.org/apiv2/packs/21147/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21147/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-01-08T08:24:49",
      "description": "",
      "id": 21098,
      "name": "20170108 - Embracer - white key chord sequence @ 100 bpm - NI Guitar Rig 5",
      "num_downloads": 13,
      "num_sounds": 73,
      "sounds": "https://freesound.org/apiv2/packs/21098/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/21098/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-01-01T16:32:49",
      "description": "",
      "id": 20979,
      "name": "20170101 - Embracer - 3D Neverending Story - CEGB Chord @ 100 bpm - NI Guitar Rig 5",
      "num_downloads": 13,
      "num_sounds": 38,
      "sounds": "https://freesound.org/apiv2/packs/20979/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/20979/",
      "username": "Jovica",
    },
    Object {
      "created": "2017-01-01T14:48:15",
      "description": "",
      "id": 20978,
      "name": "20170101 - Embracer - Cmaj Chord @ 100 bpm - 3D Orthanc - Longflow",
      "num_downloads": 12,
      "num_sounds": 38,
      "sounds": "https://freesound.org/apiv2/packs/20978/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/20978/",
      "username": "Jovica",
    },
    Object {
      "created": "2016-12-31T15:10:16",
      "description": "",
      "id": 20968,
      "name": "20161230 - Embracer - C-F-G-C Chord sequence - 100 bpm",
      "num_downloads": 16,
      "num_sounds": 32,
      "sounds": "https://freesound.org/apiv2/packs/20968/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/20968/",
      "username": "Jovica",
    },
    Object {
      "created": "2016-12-30T17:30:46",
      "description": "",
      "id": 20965,
      "name": "20161230 - Embracer - Warm Bath MW - 8 bar chord sequence - f - g - c - 120 bpm",
      "num_downloads": 23,
      "num_sounds": 20,
      "sounds": "https://freesound.org/apiv2/packs/20965/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/20965/",
      "username": "Jovica",
    },
    Object {
      "created": "2016-12-29T16:11:36",
      "description": "",
      "id": 20957,
      "name": "20161229 - Piano Chord Sequence C-Dmin-A7-Dmin-Emin-B7-Emin 120 bpm + variations",
      "num_downloads": 18,
      "num_sounds": 17,
      "sounds": "https://freesound.org/apiv2/packs/20957/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/20957/",
      "username": "Jovica",
    },
    Object {
      "created": "2016-01-17T17:24:21",
      "description": "",
      "id": 18795,
      "name": "2016 Arpeggios",
      "num_downloads": 17,
      "num_sounds": 1,
      "sounds": "https://freesound.org/apiv2/packs/18795/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/18795/",
      "username": "Jovica",
    },
    Object {
      "created": "2015-01-18T10:44:53",
      "description": "",
      "id": 16063,
      "name": "Slayers Pack 1",
      "num_downloads": 28,
      "num_sounds": 8,
      "sounds": "https://freesound.org/apiv2/packs/16063/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/16063/",
      "username": "Jovica",
    },
    Object {
      "created": "2014-09-19T18:52:10",
      "description": "",
      "id": 15303,
      "name": "20140914_attackloop_160BPM_4/4_loop_pack",
      "num_downloads": 56,
      "num_sounds": 36,
      "sounds": "https://freesound.org/apiv2/packs/15303/sounds/",
      "url": "https://freesound.org/people/Jovica/packs/15303/",
      "username": "Jovica",
    },
  ],
}
`;

exports[`API User should get a users data 3`] = `
Object {
  "count": 0,
  "next": null,
  "previous": null,
  "results": Array [],
}
`;

exports[`API User should get a users data 4`] = `
Object {
  "detail": "Invalid url",
}
`;

exports[`API User should get user 1`] = `
Object {
  "about": "Policy of use: you must state somewhere somehow (credit lines, web page, whatever) that the Freesound Project served this sounds. It is irrelevant to me whether you mention or not my authorship. Can't credit? Send me a personal message. (Thanks to dobroide for these words!)

If possible, I would also like to hear where the sounds are used, so if you can send me a link or something else, please do so. Thanks!

For some more information about me, click on the links below:
<a href=\\"http://www.c-o-l-o-u-r-s.org/\\" rel=\\"nofollow\\">http://www.c-o-l-o-u-r-s.org/</a>. If you want to support our label, please buy something over there.

Check this out: <a href=\\"https://cyclesofmoebius.bandcamp.com/\\" rel=\\"nofollow\\">https://cyclesofmoebius.bandcamp.com/</a>


Namaste!
Jovica Storer",
  "avatar": Object {
    "large": "https://freesound.org/data/avatars/2/2518_L.jpg",
    "medium": "https://freesound.org/data/avatars/2/2518_M.jpg",
    "small": "https://freesound.org/data/avatars/2/2518_S.jpg",
  },
  "bookmarkCategories": [Function],
  "bookmarkCategorySounds": [Function],
  "bookmark_categories": "https://freesound.org/apiv2/users/Jovica/bookmark_categories/",
  "date_joined": "2005-05-07T17:49:39",
  "home_page": "http://www.c-o-l-o-u-r-s.com/",
  "num_comments": 3,
  "num_packs": 239,
  "num_posts": 33,
  "num_sounds": 7813,
  "packs": [Function],
  "sounds": [Function],
  "url": "https://freesound.org/people/Jovica/",
  "username": "Jovica",
}
`;

exports[`API should get sound 1`] = `
Object {
  "analysis": "No descriptors specified. You should indicate which descriptors you want with the 'descriptors' request parameter.",
  "analysis_frames": "https://freesound.org/data/analysis/96/96541_7037_frames.json",
  "analysis_stats": "https://freesound.org/apiv2/sounds/96541/analysis/",
  "avg_rating": 4.666666666666665,
  "bitdepth": 24,
  "bitrate": 2117,
  "bookmark": "https://freesound.org/apiv2/sounds/96541/bookmark/",
  "channels": 2,
  "comment": "https://freesound.org/apiv2/sounds/96541/comment/",
  "comments": "https://freesound.org/apiv2/sounds/96541/comments/",
  "created": "2010-05-12T20:29:49",
  "description": "For the user \\"Stealth Inc.\\", my girlfriend saying \\"Coming soon\\". Recorded with an AT2020 microphone into an Apogee Duet and edited with Record.",
  "download": "https://freesound.org/apiv2/sounds/96541/download/",
  "duration": 1.8774829932,
  "filesize": 496906,
  "geotag": null,
  "id": 96541,
  "images": Object {
    "spectral_l": "https://freesound.org/data/displays/96/96541_7037_spec_L.jpg",
    "spectral_m": "https://freesound.org/data/displays/96/96541_7037_spec_M.jpg",
    "waveform_l": "https://freesound.org/data/displays/96/96541_7037_wave_L.png",
    "waveform_m": "https://freesound.org/data/displays/96/96541_7037_wave_M.png",
  },
  "license": "http://creativecommons.org/licenses/by/3.0/",
  "name": "coming soon.wav",
  "num_comments": 1,
  "num_ratings": 3,
  "pack": "https://freesound.org/apiv2/packs/6201/",
  "pack_name": "Stealth Inc.",
  "previews": Object {
    "preview-hq-mp3": "https://freesound.org/data/previews/96/96541_7037-hq.mp3",
    "preview-hq-ogg": "https://freesound.org/data/previews/96/96541_7037-hq.ogg",
    "preview-lq-mp3": "https://freesound.org/data/previews/96/96541_7037-lq.mp3",
    "preview-lq-ogg": "https://freesound.org/data/previews/96/96541_7037-lq.ogg",
  },
  "rate": "https://freesound.org/apiv2/sounds/96541/rate/",
  "samplerate": 44100,
  "similar_sounds": "https://freesound.org/apiv2/sounds/96541/similar/",
  "tags": Array [
    "request",
    "female",
    "girl",
    "talk",
    "vocal",
    "voice",
    "american",
    "woman",
    "english",
    "speak",
  ],
  "type": "wav",
  "url": "https://freesound.org/people/tim.kahn/sounds/96541/",
  "username": "tim.kahn",
}
`;

exports[`API should get user 1`] = `
Object {
  "about": "Policy of use: you must state somewhere somehow (credit lines, web page, whatever) that the Freesound Project served this sounds. It is irrelevant to me whether you mention or not my authorship. Can't credit? Send me a personal message. (Thanks to dobroide for these words!)

If possible, I would also like to hear where the sounds are used, so if you can send me a link or something else, please do so. Thanks!

For some more information about me, click on the links below:
<a href=\\"http://www.c-o-l-o-u-r-s.org/\\" rel=\\"nofollow\\">http://www.c-o-l-o-u-r-s.org/</a>. If you want to support our label, please buy something over there.

Check this out: <a href=\\"https://cyclesofmoebius.bandcamp.com/\\" rel=\\"nofollow\\">https://cyclesofmoebius.bandcamp.com/</a>


Namaste!
Jovica Storer",
  "avatar": Object {
    "large": "https://freesound.org/data/avatars/2/2518_L.jpg",
    "medium": "https://freesound.org/data/avatars/2/2518_M.jpg",
    "small": "https://freesound.org/data/avatars/2/2518_S.jpg",
  },
  "bookmarkCategories": [Function],
  "bookmarkCategorySounds": [Function],
  "bookmark_categories": "https://freesound.org/apiv2/users/Jovica/bookmark_categories/",
  "date_joined": "2005-05-07T17:49:39",
  "home_page": "http://www.c-o-l-o-u-r-s.com/",
  "num_comments": 3,
  "num_packs": 239,
  "num_posts": 33,
  "num_sounds": 7813,
  "packs": [Function],
  "sounds": [Function],
  "url": "https://freesound.org/people/Jovica/",
  "username": "Jovica",
}
`;

exports[`API should text search 1`] = `
Object {
  "count": 374921,
  "getItem": [Function],
  "getSound": [Function],
  "next": "http://freesound.org/apiv2/search/text/?&query=&page=2",
  "nextPage": [Function],
  "previous": null,
  "previousPage": [Function],
  "results": Array [
    Object {
      "id": 341293,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "RobotFX2.mp3",
      "tags": Array [
        "Sound",
        "Clash",
        "Arcade",
        "Robot",
        "Glitch",
        "Computer",
        "Pitched",
      ],
      "username": "Bird_man",
    },
    Object {
      "id": 341297,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 02",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341298,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 001",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341300,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Close 01",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341302,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Locked 02",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "locked",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341303,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Locked 03",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "locked",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341304,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 03",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341305,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 04",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341306,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 05",
      "tags": Array [
        "handle",
        "Door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341307,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 06",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341308,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 07",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341309,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 08",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341310,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 09",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341311,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 10",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
    Object {
      "id": 341312,
      "license": "http://creativecommons.org/publicdomain/zero/1.0/",
      "name": "Door Handle 11",
      "tags": Array [
        "handle",
        "door",
        "field-recording",
        "open",
        "close",
      ],
      "username": "ppfpower87",
    },
  ],
}
`;
